src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2383,7 +2383,7 @@ private BoundExpression BindUnaryOperatorCore(CSharpSyntaxNode node, string oper
                 underlyingType :
                 GetSpecialType(upconvertSpecialType, diagnostics, syntax);
 
-            newOperand = CreateConversion(newOperand, upconvertType, diagnostics);
+            _ = CreateConversion(newOperand, upconvertType, diagnostics);
```

Comment:
also surprising!

Reply:
Confirmed `newOperand` indeed has no further references in this method!